wpimath/src/main/java/edu/wpi/first/math/filter/SlewRateLimiter.java

```diff
@@ -14,53 +14,97 @@
  * edu.wpi.first.math.trajectory.TrapezoidProfile} instead.
  */
 public class SlewRateLimiter {
-  private final double m_rateLimit;
-  private double m_prevVal;
-  private double m_prevTime;
+    private double m_prevVal;
+    private double m_prevTime;
+    private double m_forwardRateLimit;
+    private double m_backwardRateLimit;
 
-  /**
-   * Creates a new SlewRateLimiter with the given rate limit and initial value.
-   *
-   * @param rateLimit The rate-of-change limit, in units per second.
-   * @param initialValue The initial value of the input.
-   */
-  public SlewRateLimiter(double rateLimit, double initialValue) {
-    m_rateLimit = rateLimit;
-    m_prevVal = initialValue;
-    m_prevTime = WPIUtilJNI.now() * 1e-6;
-  }
+    /**
+     * Creates a new SlewRateLimiter with the given rate limit and initial value.
+     * Sets the given rate limit to both the forward and backward limits.
+     *
+     * @param rateLimit    The rate-of-change limit, in units per second.
+     * @param initialValue The initial value of the input.
+     */
+    public SlewRateLimiter(double rateLimit, double initialValue) {
+        m_forwardRateLimit = rateLimit;
+        m_backwardRateLimit = rateLimit;
+        m_prevVal = initialValue;
+        m_prevTime = WPIUtilJNI.now() * 1e-6;
+    }
 
-  /**
-   * Creates a new SlewRateLimiter with the given rate limit and an initial value of zero.
-   *
-   * @param rateLimit The rate-of-change limit, in units per second.
-   */
-  public SlewRateLimiter(double rateLimit) {
-    this(rateLimit, 0);
-  }
+    /**
+     * Creates a new SlewRateLimiter with the given rate limit and an initial value of the given value.
```

Comment:
There's no initial value in this constructor (and you probably don't want a version of this constructor with an initial value because it'd have the same signature as the constructor taking `(double forwardLimit, double backwardLimit)`)